site/app/views/MiscView.php

```diff
@@ -11,20 +11,25 @@ public function displayFile($file_contents) {
     }
 
     public function displayCode($file_type, $filename, $file_contents) {
-        $this->core->getOutput()->addVendorCss(FileUtils::joinPaths('codemirror', 'codemirror.css'));
-        $this->core->getOutput()->addVendorCss(FileUtils::joinPaths('codemirror', 'theme', 'eclipse.css'));
-        $this->core->getOutput()->addVendorCss(FileUtils::joinPaths('codemirror', 'theme', 'monokai.css'));
-        $this->core->getOutput()->addVendorJs(FileUtils::joinPaths('jquery', 'jquery.min.js'));
-        $this->core->getOutput()->addVendorJs(FileUtils::joinPaths('codemirror', 'codemirror.js'));
-        $this->core->getOutput()->addVendorJs(FileUtils::joinPaths('codemirror', 'mode', 'clike', 'clike.js'));
-        $this->core->getOutput()->addVendorJs(FileUtils::joinPaths('codemirror', 'mode', 'python', 'python.js'));
-        $this->core->getOutput()->addVendorJs(FileUtils::joinPaths('codemirror', 'mode', 'shell', 'shell.js'));
+        $code_css = array();
+        $code_css[] = $this->core->getOutput()->timestampResource(FileUtils::joinPaths('codemirror', 'codemirror.css'), 'vendor');
```

Comment:
How about using `addInternalCss`, `addVendorCss`, `addInternalJs` and `addVendorJs`?

Reply:
Sorry I just saw you deleted those lines.

Reply:
The `add` functions load stuff into global header, but this is a partial module which is conditionally loaded onto the page, after the header is made. For those scenarios, we use the `timestamp` function to format.

Reply:
Thank you for your explanation. :)